src/main/c/com/micromouse/robot/robot.c

```diff
@@ -182,4 +182,60 @@ void RobotDestructor(Robot* winslow) {
 
 	// Free up the memory for the Robot
 	free(winslow);
-}
\ No newline at end of file
+}
+
+// Robot Location -> [Linked-list of [Moves]]
+// Returns a linked list of moves for the robot to execute, starts by 
+// searching the maze for the path to the goal, and returns the path as a 
+// Linked-list of Moves.
+Cell* SolveMaze(Robot* winslow, Location *goal) {
+  // Create path to return and load starting location into path
+  Location* curr_loc = winslow->location_;
+
+  // Mark curr_loc as visited
+  int curr_x = curr_loc->x;
+  int curr_y = curr_loc->y;
+  List* queue = InitializeList(winslow->maze_[curr_x][curr_y]);
+  Cell* curr_cell = winslow->maze_[curr_x][curr_y];
+  curr_cell->visited = 1;
+
+  int is_queue_empty = empty(queue);
+  while(!is_queue_empty) {
+    // TODO: Dequeue curr_cell from queue.
+    curr_cell = Front(&queue);
+    if(curr_cell->location == goal) {
```

Comment:
this probably does not do what you think it will do. You cannot just compare structs with == and have it check their members. This is probably just comparing their pointers so it will only be equal if they are the same pointer.